### PR TITLE
Implement limiting the number of entries

### PR DESCRIPTION
**Pull request by NelsonVides**

### CODE REVIEW

File: src/segmented_cache_helpers.erl [hunk 1]
```diff
@@ -268,31 +300,33 @@ assert_parameters(Opts) when is_map(Opts) ->
                {hours, H} -> timer:hours(H);
                T when is_integer(T) -> timer:minutes(T)
            end,
+    true = is_integer(N) andalso N > 0,
+    true = (EntriesLimit =:= infinity) orelse (is_integer(EntriesLimit) andalso N > 0),
```
Comment (JanuszJakubiec) [inline src/segmented_cache_helpers.erl:304]:
```suggestion
    true = (EntriesLimit =:= infinity) orelse (is_integer(EntriesLimit) andalso EntriesLimit > 0),
```

File: src/segmented_cache_helpers.erl [hunk 2]
```diff
@@ -268,31 +300,33 @@ assert_parameters(Opts) when is_map(Opts) ->
...
+-if(?OTP_RELEASE >= 25).
+ets_settings(#{entries_limit := infinity}) ->
+    [set, public,
+     {read_concurrency, true},
+     {write_concurrency, auto}];
+ets_settings(#{entries_limit := _}) ->
+    [set, public,
+     {read_concurrency, true},
+     {write_concurrency, true}].
+-elif(?OTP_RELEASE >= 21).
```
Comment (JanuszJakubiec) [inline src/segmented_cache_helpers.erl:328]: Why is there `elif`, if in rebar.config, the minimum_otp_vsn is set to 21? Shouldn't it be enough to use `-else`? Or is it added to increase readability?

File: src/segmented_cache_helpers.erl (hunk 1 above)
Comment (NelsonVides) [reply src/segmented_cache_helpers.erl:304]: Ah, the issues of copy-pasting 🙈

File: src/segmented_cache_helpers.erl (hunk 2 above)
Comment (NelsonVides) [reply src/segmented_cache_helpers.erl:328]: No real reason, this just came from the previous ifdef.

**Review by JanuszJakubiec — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**